src/nvim/decoration.h

```diff
@@ -76,11 +91,22 @@ typedef struct {
   bool running_decor_provider;
 } DecorState;
 
-EXTERN DecorState decor_state INIT( = { 0 });
+EXTERN DecorState decor_state INIT( = { .free_slot_i = -1 });
 // TODO(bfredl): These should maybe be per-buffer, so that all resources
 // associated with a buffer can be freed when the buffer is unloaded.
 EXTERN kvec_t(DecorSignHighlight) decor_items INIT( = KV_INITIAL_VALUE);
 
 #ifdef INCLUDE_GENERATED_DECLARATIONS
 # include "decoration.h.generated.h"
 #endif
+
+inline int decor_redraw_col(win_T *wp, int col, int win_col, bool hidden, DecorState *state)
+  REAL_FATTR_ALWAYS_INLINE;
+
+inline int decor_redraw_col(win_T *wp, int col, int win_col, bool hidden, DecorState *state)
```

Comment:
```suggestion
# include "decoration.h.inline.generated.h"
#endif

inline int decor_redraw_col(win_T *wp, int col, int win_col, bool hidden, DecorState *state)
  FUNC_ATTR_ALWAYS_INLINE
```

Reply:
I made the function `static inline` because otherwise "decoration.h.inline.generated.h" is empty and linking fails with unresolved symbol error.